common/src/main/java/org/wildfly/swarm/ts/common/DockerRunner.java

```diff
@@ -0,0 +1,167 @@
+package org.wildfly.swarm.ts.common;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.UUID;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+import java.util.logging.ConsoleHandler;
+import java.util.logging.LogRecord;
+import java.util.logging.Logger;
+import java.util.logging.SimpleFormatter;
+
+/**
+ * Simple container runner using "docker" command
+ * @author juagonza
+ *
+ */
+public class DockerRunner {
+
+    private static Logger LOGGER = null;
+    private String containerId;
+    private String containerImage;
+    private List<String> ports = new ArrayList<>();
+    private Map<String, String> environmentVariables = new HashMap<>();
+    private String logWait;
+    private long timeoutMillis = 60000;
+
+    public DockerRunner() {
+        LOGGER = Logger.getLogger(DockerRunner.class.getName());
+        LOGGER.setUseParentHandlers(false);
+
+        CustomDockerRunnerFormatter formatter = new CustomDockerRunnerFormatter();
+        ConsoleHandler handler = new ConsoleHandler();
+        handler.setFormatter(formatter);
+
+        LOGGER.addHandler(handler);
+    }
+
+    public DockerRunner(String containerImage, String logWait) {
```

Comment:
`containerImage` should be mandatory, and `logWait` should be a builder method. That is, only one constructor, with one parameter.

Reply:
Ah one more mandatory parameter: container ID. Docker would assign an artificial one automatically, but I think all callers should always provide a human-readable & meaningful one.